src/common/k8s-api/endpoints/ingress.api.ts

```diff
@@ -102,13 +103,13 @@ export interface Ingress {
     // extensions/v1beta1
     backend?: IExtensionsBackend;
     // networking.k8s.io/v1
-    defaultBackend?: INetworkingBackend & {
+    defaultBackend?: RequireExactlyOne<INetworkingBackend & {
```

Comment:
what does this do? There is no KeysType parameter

Reply:
`RequireExactlyOne` comes from the `type-fest` package. Given a type `T` that has some fields, `RequireExactlyOne<T>` means that at most one and only one of the fields must be set.

Reply:
So either `service` or `resource`? This is not at all obvious. I couldn't even see from the type-fest docs that you can omit the second param (KeysType). Oh, I see, default is `keyof ObjectType`.  Makes me think of boost. Can't decipher it either.

Reply:
Would you prefer if I did it manually? That type is equivalent to the following:

```ts
type DefaultBackend = {
  service: IIngressService;
  resource?: undefined;
} & {
  service?: undefined;
  resource: {
    apiGroup: string;
    kind: string;
    name: string;
  }
};
```

Reply:
Only change it if you think it is better. Maybe just a comment would help clarify?

Reply:
I'll leave a comment